plugin/build.gradle

```diff
@@ -417,6 +417,8 @@ configurations.all {
     resolutionStrategy.force "org.apache.logging.log4j:log4j-api:2.24.2"
     resolutionStrategy.force "org.apache.logging.log4j:log4j-core:2.24.2"
     resolutionStrategy.force "jakarta.json:jakarta.json-api:2.1.3"
+    resolutionStrategy.force "com.google.code.gson:gson:2.11.0"
```

Comment:
other modules are using `2.10.1`. Should we upgrade all of them? Also in this file, we have gson in line 84 with 2.10.1, may be we can remove this?

Otherwise we will keep adding versions this way, which is not a clean code IMO.

Reply:
Yeah, agree. Let me upgrade all to 2.11.0.

Reply:
I was suggesting something like this: https://github.com/opensearch-project/ml-commons/pull/3478/files